src/api_helpers.jl

```diff
@@ -0,0 +1,297 @@
+    return threshold[], alignment[]
+end
+
+function h5p_get_alloc_time(plist_id)
+    alloc_time = Ref{Cint}()
+    h5p_get_alloc_time(plist_id, alloc_time)
+    return alloc_time[]
+end
+
+function h5p_get_char_encoding(plist_id)
+    encoding = Ref{Cint}()
+    h5p_get_char_encoding(plist_id, encoding)
+    return encoding[]
+end
+
+function h5p_get_chunk(plist_id)
+    n = h5p_get_chunk(plist_id, 0, C_NULL)
+    cdims = Vector{hsize_t}(undef, n)
+    h5p_get_chunk(plist_id, n, cdims)
+    return cdims
+end
+
+function h5p_get_create_intermediate_group(plist_id)
+    cig = Ref{Cuint}()
+    h5p_get_create_intermediate_group(plist_id, cig)
+    return cig[]
+end
+
+function h5p_get_dxpl_mpio(dxpl_id)
+    xfer_mode = Ref{Cint}()
+    h5p_get_dxpl_mpio(dxpl_id, xfer_mode)
+    return xfer_mode[]
+end
+
+function h5p_get_fclose_degree(fapl_id)
+    out = Ref{Cint}()
+    h5p_get_fclose_degree(fapl_id, out)
+    return out[]
+end
+
+function h5p_get_libver_bounds(plist_id)
+    low = Ref{Cint}()
+    high = Ref{Cint}()
+    h5p_get_libver_bounds(plist_id, low, high)
+    return low[], high[]
+end
+
+function h5p_get_local_heap_size_hint(plist_id)
+    size_hint = Ref{Csize_t}()
+    h5p_get_local_heap_size_hint(plist_id, size_hint)
+    return size_hint[]
+end
+
+function h5p_get_obj_track_times(plist_id)
+    track_times = Ref{UInt8}()
+    h5p_get_obj_track_times(plist_id, track_times)
+    return track_times[] != 0x0
+end
+
+function h5p_get_userblock(plist_id)
+    len = Ref{hsize_t}()
+    h5p_get_userblock(plist_id, len)
+    return len[]
+end
+
+###
+### Reference Interface
+###
+
+###
+### Dataspace Interface
+###
+
+function h5s_get_simple_extent_dims(space_id)
+    n = h5s_get_simple_extent_ndims(space_id)
+    dims = Vector{hsize_t}(undef, n)
+    maxdims = Vector{hsize_t}(undef, n)
+    h5s_get_simple_extent_dims(space_id, dims, maxdims)
+    return dims, maxdims
+end
+
+
+###
+### Datatype Interface
+###
+
+function h5t_get_array_dims(type_id)
+    nd = h5t_get_array_ndims(type_id)
+    dims = Vector{hsize_t}(undef, nd)
+    h5t_get_array_dims(type_id, dims)
+    return dims
+end
+
+# Note: The following two functions implement direct ccalls because the binding generator
+# cannot (yet) do the string wrapping and memory freeing.
+function h5t_get_member_name(type_id, index)
+    pn = ccall((:H5Tget_member_name, libhdf5), Ptr{UInt8}, (hid_t, Cuint), type_id, index)
+    if pn == C_NULL
+        error("Error getting name of compound datatype member #", index)
+    end
+    s = unsafe_string(pn)
+    h5_free_memory(pn)
+    return s
+end
+
+function h5t_get_tag(type_id)
+    pc = ccall((:H5Tget_tag, libhdf5), Ptr{UInt8}, (hid_t,), type_id)
+    if pc == C_NULL
+        error("Error getting opaque tag")
+    end
+    s = unsafe_string(pc)
+    h5_free_memory(pc)
+    return s
+end
+
+###
+### Optimized Functions Interface
+###
+
+###
+### HDF5 Lite Interface
+###
+
+function h5lt_dtype_to_text(dtype_id)
```

Comment:
unrelated, but lol this is such a hack, there's got to be a better way of doing this somehow.